### PR TITLE
Use modern cmake function export to generate target

### PR DESCRIPTION
**Pull request by JackBoosY**
Use modern cmake function `export` to generate targets instead of manually declaring targets.
Remove useless cmake code and macros.

### CODE REVIEW

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/rapidjson?pullRequest=1901) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/rapidjson?pullRequest=1901) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/rapidjson?pullRequest=1901) it.</sub>

File: RapidJSONConfig.cmake.in [hunk 1]
```diff
@@ -14,12 +14,6 @@ set( RapidJSON_DIR "@CONFIG_DIR@")
...
-set( RapidJSON_INCLUDE_DIRS  "@RapidJSON_INCLUDE_DIR@" )
-message(STATUS "RapidJSON found. Headers: ${RapidJSON_INCLUDE_DIRS}")
+get_target_property(RapidJSON_INCLUDE_DIR RapidJSON INTERFACE_INCLUDE_DIRECTORIES)
 
-if(NOT TARGET rapidjson)
-  add_library(rapidjson INTERFACE IMPORTED)
-  set_property(TARGET rapidjson PROPERTY
-    INTERFACE_INCLUDE_DIRECTORIES ${RapidJSON_INCLUDE_DIRS})
-endif()
+set( RapidJSON_INCLUDE_DIRS ${RapidJSON_INCLUDE_DIR} )
```
Comment (JackBoosY) [inline RapidJSONConfig.cmake.in:19]: For compatibility, we still need to keep it.

**Comment by igor-sirotin** [comment]
In case adding RapidJSON with `add_subdirectory` (e.g. via `FetchContent`), the source `include` directory  should also be marked as includedir:

```
target_include_directories(RapidJSON INTERFACE
        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include> # <-- Added this line
        $<INSTALL_INTERFACE:include/rapidjson>)
```

**Comment by JackBoosY** [comment]
@igor-sirotin Now it's already added.

**Comment by MrJia1997** [comment]
Is there anything block for this PR?

**Comment by JackBoosY** [comment]
@MrJia1997 I don't know, this PR is ready for review and merge now.

**Comment by cthulhu-irl** [comment]
Hi. I was about to make a similar MR. now I'm wondering why it hasn't been merged into codebase.

may I ask if there is anything keeping it from being merged?

CC @miloyip @igor-sirotin

**Comment by dotnwat** [comment]
Please merge this.

**Comment by AVividLight** [comment]
Hello,

I'm trying to use CMake's FetchContent to add RapidJSON to my project, but with the following in my CMakeLists.txt I get the below error when configuring:

```CMake
find_package(RapidJSON QUIET)
if(NOT ${RapidJSON_FOUND})
	message(STATUS "RapidJSON will be fetched.")

	FetchContent_Declare(external-RapidJSON
		GIT_REPOSITORY	https://github.com/Tencent/rapidjson.git
		GIT_TAG			476ffa2fd272243275a74c36952f210267dc3088 #Latest commit as of 8/23/2023
		GIT_PROGRESS	TRUE
		OVERRIDE_FIND_PACKAGE
	)
	set(RAPIDJSON_BUILD_DOC OFF CACHE BOOL "" FORCE)
	set(RAPIDJSON_BUILD_EXAMPLES OFF CACHE BOOL "" FORCE)
	set(RAPIDJSON_BUILD_TESTS OFF CACHE BOOL "" FORCE)
	set(RAPIDJSON_BUILD_THIRDPARTY_GTEST OFF CACHE BOOL "" FORCE)
	set(RAPIDJSON_BUILD_CXX17 ON CACHE BOOL "" FORCE)
	set(RAPIDJSON_BUILD_ASAN OFF CACHE BOOL "" FORCE)
	set(RAPIDJSON_BUILD_UBSAN OFF CACHE BOOL "" FORCE)
	set(RAPIDJSON_ENABLE_INSTRUMENTATION_OPT ON CACHE BOOL "" FORCE)
	set(RAPIDJSON_HAS_STDSTRING ON CACHE BOOL "" FORCE)
	set(RAPIDJSON_USE_MEMBERSMAP OFF CACHE BOOL "" FORCE)

	FetchContent_MakeAvailable(external-RapidJSON)
	find_package(external-RapidJSON)
endif(NOT ${RapidJSON_FOUND})

...

target_link_libraries(RPGEngine PRIVATE
	RapidJSON
	)
```

```
CMake Error at E:/Developer/Builds/RPGEngine/_deps/external-rapidjson-build/RapidJSONConfig.cmake:3 (include):
  include could not find requested file:

    E:/Developer/Builds/RPGEngine/_deps/external-rapidjson-build/RapidJSON-targets.cmake
Call Stack (most recent call first):
  CMakeLists.txt:62 (find_package) {line 64 is the first line in the code posted above}


CMake Error at E:/Developer/Builds/RPGEngine/_deps/external-rapidjson-build/RapidJSONConfig.cmake:17 (get_target_property):
  get_target_property() called with non-existent target "RapidJSON".
Call Stack (most recent call first):
  CMakeLists.txt:62 (find_package)


Configuring incomplete, errors occurred!
```

I am not a CMake expert, and honestly FetchContent seems like magic to me, so I may be doing something wrong. Do you see what mistake I've made, or do you have an example of how to utilize the changes introduced in this pull request? Here is my full code: https://github.com/AVividLight/RPGEngine/blob/master/CMakeLists.txt#L62

Thank you very much!

**Pull request merged**